src/Mod/PartDesign/App/Feature.cpp

```diff
@@ -100,6 +101,33 @@ TopoDS_Shape Feature::getSolid(const TopoDS_Shape& shape)
     return {};
 }
 
+TopoShape Feature::getSolid(const TopoShape& shape, bool force)
+{
+    if (shape.isNull()) {
+        throw Part::NullShapeException("Null shape");
+    }
+    int count = shape.countSubShapes(TopAbs_SOLID);
+    if (count > 1) {
+        if (getFeatureBody()) {
+            auto res = shape;
+            res.fixSolidOrientation();
+            return res;
+        }
+        throw Base::RuntimeError(
+            "Result has multiple solids.\n"
+            "To allow multiple solids, please set 'SingleSolid' property of the body to false");

```

Comment:
This is not accurate in main, we don't support multiple solids at all. (unless something big changed and I missed it!!)